pkg/server/server.go

```diff
@@ -775,6 +773,7 @@ func NewServer(cfg Config, stopper *stop.Stopper) (*Server, error) {
 	kvserver.RegisterPerReplicaServer(grpcServer.Server, node.perReplicaServer)
 	kvserver.RegisterPerStoreServer(grpcServer.Server, node.perReplicaServer)
 	ctpb.RegisterSideTransportServer(grpcServer.Server, ctReceiver)
+	gossip.RegisterGossipServer(grpcServer.Server, g)
```

Comment:
This is probably fine, but is there a risk that starting the gossip server this late during startup could cause issues since gossip is used during bootstrapping?

Reply:
It does work down here, but I agree it is a little riskier. I moved this up to right after grpcServer construction.